src/decoder/lattice-faster-decoder.cc

```diff
@@ -809,8 +809,11 @@ void LatticeFasterDecoder::ProcessNonemitting(BaseFloat cutoff) {
   // problem did not improve overall speed.
 
   KALDI_ASSERT(queue_.empty());
-  for (const Elem *e = toks_.GetList(); e != NULL;  e = e->tail)
-    queue_.push_back(e->key);
+  for (const Elem *e = toks_.GetList(); e != NULL;  e = e->tail) {
+    StateId key = e->key;
```

Comment:
actually, I think it would be clearer if this variable were named state, and not key.  I know it was key in my branch; I'll fix the conflict later.